File: Wander.Server/Scripts/Game/SoundManager.ts
class SoundManager {

    dayAmbientSound: Phaser.Sound;
    nightAmbientSound: Phaser.Sound;
    rainAmbientSound: Phaser.Sound;
    footstepSound: Phaser.Sound;
    actionSound: Phaser.Sound;
    chatSound: Phaser.Sound;
    game: Phaser.Game;
    state: GameState;

    constructor(game: Phaser.Game, state: GameState) {
        this.game = game;
        this.state = state;
    }

    preload() {
        this.game.load.audio("dayAmbient", ["Content/Game/Sounds/ambient-day.mp3", "Content/Game/Sounds/ambient-day.ogg"]);
        this.game.load.audio("nightAmbient", ["Content/Game/Sounds/ambient-night.mp3", "Content/Game/Sounds/ambient-night.ogg"]);
        this.game.load.audio("rainAmbient", ["Content/Game/Sounds/ambient-rain.mp3", "Content/Game/Sounds/ambient-rain.ogg"]);
        this.game.load.audio("action", "Content/Game/Sounds/action.ogg");
        this.game.load.audio("chat", "Content/Game/Sounds/chat.ogg");
        if (currentState.houseId == -1) {
            this.game.load.audio("footstep", "Content/Game/Sounds/footstep.ogg");
        } else {
            this.game.load.audio("footstep", "Content/Game/Sounds/footstepHouse.wav");

        }
    }

    create() {
        this.dayAmbientSound = this.game.add.audio("dayAmbient");
        this.nightAmbientSound = this.game.add.audio("nightAmbient");
        this.rainAmbientSound = this.game.add.audio("rainAmbient");

        this.dayAmbientSound.loop = true;
        this.nightAmbientSound.loop = true;
        this.rainAmbientSound.loop = true;

        this.dayAmbientSound.volume = 0;
        this.nightAmbientSound.volume = 0;
        this.rainAmbientSound.volume = 0;

        this.footstepSound = this.game.add.audio("footstep");

        this.actionSound = this.game.add.audio("action");
        this.actionSound.volume = 0.3;
        this.chatSound = this.game.add.audio("chat");
        this.chatSound.volume = 0.3;
    }

    playFootStep(player: Player) {
        if (currentState.map.currentPlayer == player) {
           
            this.footstepSound.volume = 1;
        } else {
            var dist = this.game.physics.arcade.distanceBetween(player.texture, currentState.map.currentPlayer.texture);
            var maxHearingDist = this.game.camera.width / 2;
            var t = 1 - (1 / (maxHearingDist)) * dist;
            t = t < 0 ? 0 :t;
            this.footstepSound.volume = t > 1 ? 1 : t;
        }
        this.footstepSound.play();
    }

    playActionSound() {
        this.actionSound.play();
    }

    playChatSound() {
        this.chatSound.play();
    }


    update() {
        if (currentState.houseId !== -1) {
            return;
        }
        if (this.state.dayNightCycle.isDay) {
            if (!this.dayAmbientSound.isPlaying) this.dayAmbientSound.play();           
            this.nightAmbientSound.volume = Lerp(0, this.nightAmbientSound.volume, 0.01);
            this.dayAmbientSound.volume = Lerp(1, this.dayAmbientSound.volume, 0.01);
            if (this.nightAmbientSound.volume <= 0 && this.nightAmbientSound.isPlaying) this.nightAmbientSound.stop();

        } else {
            if (!this.nightAmbientSound.isPlaying) this.nightAmbientSound.play();   
            this.nightAmbientSound.volume = Lerp(1, this.nightAmbientSound.volume, 0.01);
            this.dayAmbientSound.volume = Lerp(0, this.dayAmbientSound.volume, 0.01);
            if (this.dayAmbientSound.volume <= 0 && this.dayAmbientSound.isPlaying) this.dayAmbientSound.stop();
        }

        if (this.state.isRaining()) {
            if (!this.rainAmbientSound.isPlaying) this.rainAmbientSound.play();
            this.rainAmbientSound.volume = Lerp(1, this.rainAmbientSound.volume, 0.01);
        } else {
            this.rainAmbientSound.volume = Lerp(0, this.rainAmbientSound.volume, 0.01);
            if (this.rainAmbientSound.volume <= 0) {
                this.rainAmbientSound.stop();
            }
        }
    }
}

File: Wander.Android.App/core/src/com/wander/game/weather/SoundManager.java
package com.wander.game.weather;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Music;
import com.wander.game.MainGame;
import com.wander.game.util;

/**
 * Created by val on 12/12/2015.
 */
public class SoundManager {
    private Music dayAmbientSound;
    private Music nightAmbientSound;
    private Music rainAmbientSound;
    private MainGame game;

    public SoundManager(MainGame game)
    {
        this.game = game;
        this.dayAmbientSound = Gdx.audio.newMusic(Gdx.files.internal("sounds/ambient-day.mp3"));
        this.nightAmbientSound = Gdx.audio.newMusic(Gdx.files.internal("sounds/ambient-night.mp3"));
        this.rainAmbientSound = Gdx.audio.newMusic(Gdx.files.internal("sounds/ambient-rain.mp3"));

        this.dayAmbientSound.setVolume(0);
        this.nightAmbientSound.setVolume(0);
        this.rainAmbientSound.setVolume(0);

        this.dayAmbientSound.setLooping(true);
        this.nightAmbientSound.setLooping(true);
        this.rainAmbientSound.setLooping(true);
    }

    public void update(){

        if (this.game.getScreenManager().getGameScreen().getMap().getAmbientManager().isDay()) {
            if (!this.dayAmbientSound.isPlaying()) this.dayAmbientSound.play();
            this.nightAmbientSound.setVolume(util.Lerp(0, this.nightAmbientSound.getVolume(), 0.01f));
            this.dayAmbientSound.setVolume(util.Lerp(1, this.dayAmbientSound.getVolume(), 0.01f));
            if (this.nightAmbientSound.getVolume() <= 0 && this.nightAmbientSound.isPlaying()) this.nightAmbientSound.stop();

        } else {
            if (!this.nightAmbientSound.isPlaying()) this.nightAmbientSound.play();
            this.nightAmbientSound.setVolume(util.Lerp(1, this.nightAmbientSound.getVolume(), 0.01f));
            this.dayAmbientSound.setVolume(util.Lerp(0, this.dayAmbientSound.getVolume(), 0.01f));
            if (this.dayAmbientSound.getVolume() <= 0 && this.dayAmbientSound.isPlaying()) this.dayAmbientSound.stop();
        }

        if (this.game.getScreenManager().getGameScreen().getMap().getAmbientManager().isRaining()) {
            if (!this.rainAmbientSound.isPlaying()) this.rainAmbientSound.play();
            this.rainAmbientSound.setVolume(util.Lerp(1f, this.rainAmbientSound.getVolume(), 0.01f));
        } else {
            this.rainAmbientSound.setVolume(util.Lerp(0, this.rainAmbientSound.getVolume(), 0.01f));
            if (this.rainAmbientSound.getVolume() <= 0) {
                this.rainAmbientSound.stop();
            }
        }
    }

    public void dispose(){
        this.dayAmbientSound.dispose();
        this.nightAmbientSound.dispose();
        this.rainAmbientSound.dispose();
    }
}
